build/Shared/NoAllocNewLineStreamWriter.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+
+namespace NuGet;
+
+internal class NoAllocNewLineStreamWriter : StreamWriter
+{
+    public NoAllocNewLineStreamWriter(Stream stream, Encoding encoding, int bufferSize, bool leaveOpen) :
+        base(stream, encoding, bufferSize, leaveOpen)
+    {
+
+    }
+
+    public NoAllocNewLineStreamWriter(Stream stream) :
+        base(stream)
+    {
+
+    }
+
+    private string? _newLine;
+
+    [AllowNull]
+    public override string NewLine
+    {
+        get
+        {
+            if (_newLine == null)
+            {
+                _newLine = new string(CoreNewLine);
+            }
+
+            return _newLine;
+        }
+        set
+        {
+            if (value == null)
+            {
+                value = "\r\n";
```

Comment:
```suggestion
                value = Environment.NewLine;
```

`\r\n` is wrong on Linux and Mac, and some people do run the .NET Framework version on Linux and Mac through mono.

Reply:
I've addressed this